### PR TITLE
bpo-32035: Update zipfile doc

### PR DESCRIPTION
**Pull request by srcecde**
Updated the cpython/Doc/library/zipfile.rst documentation.

https://bugs.python.org/issue32035

### CODE REVIEW

**Comment by the-knights-who-say-ni** [comment]
Hello, and thanks for your contribution!

I'm a bot set up to make sure that the project can legally accept your contribution by verifying you have signed the [PSF contributor agreement](https://www.python.org/psf/contrib/contrib-form/) (CLA).

Unfortunately our records indicate you have not signed the CLA. For legal reasons we need you to sign this before we can look at your contribution. Please follow [the steps outlined in the CPython devguide](https://devguide.python.org/pullrequest/#licensing) to rectify this issue.


Thanks again to your contribution and we look forward to looking at it!

File: Doc/library/zipfile.rst [hunk 1]
```diff
@@ -385,9 +385,9 @@ ZipFile Objects
 
 .. method:: ZipFile.writestr(zinfo_or_arcname, data[, compress_type])
 
-   Write the string *data* to the archive; *zinfo_or_arcname* is either the file
-   name it will be given in the archive, or a :class:`ZipInfo` instance.  If it's
-   an instance, at least the filename, date, and time must be given.  If it's a
+   Write the string or bytes *data* to the archive; if it is a *str*, it should be encoded as UTF-8.
+   *zinfo_or_arcname* is either the name of the file in the archive or a :class:`ZipInfo` instance.
```
Comment (vadmium) [inline Doc/library/zipfile.rst]: The doc string wording works better IMO; see <https://hg.python.org/cpython/rev/19196cce1431>. In particular “_bytes_ instance” makes more sense than “the bytes _data_”, and “it is encoded” is clearer; “it should be encoded” sounds like the responsibility of the caller.

File: Doc/library/zipfile.rst [hunk 2]
```diff
@@ -385,10 +385,11 @@ ZipFile Objects
 
 .. method:: ZipFile.writestr(zinfo_or_arcname, data[, compress_type])
 
-   Write the string *data* to the archive; *zinfo_or_arcname* is either the file
-   name it will be given in the archive, or a :class:`ZipInfo` instance.  If it's
-   an instance, at least the filename, date, and time must be given.  If it's a
-   name, the date and time is set to the current date and time.
+   Write a file into archive. The content is *data* which may be either a *str* or a *bytes* instance;
```
Comment (serhiy-storchaka) [inline Doc/library/zipfile.rst]: Isn't this line too long?

`ZipFile.write()` writes a file into archive. `ZipFile.writestr()` writes a bytes data or a string.

File: Doc/library/zipfile.rst [hunk 3]
```diff
@@ -385,10 +385,11 @@ ZipFile Objects
...
+   if it is a *str*, it is encoded as UTF-8.
```
Comment (serhiy-storchaka) [inline Doc/library/zipfile.rst]: I'm not sure this looks correct. `str` instance is not encoded. It is just a sequence of character, and how it is represented internally is an implementation detail.

File: Doc/library/zipfile.rst [hunk 4]
```diff
@@ -385,10 +385,11 @@ ZipFile Objects
...
+   *zinfo_or_arcname* is either the name of the file in the archive or a :class:`ZipInfo` instance.
```
Comment (serhiy-storchaka) [inline Doc/library/zipfile.rst]: Please avoid unneeded rewrapping. Wrap only long lines that you have changed in any way.

File: Doc/library/zipfile.rst (hunk 3 above)
Comment (srcecde) [reply Doc/library/zipfile.rst]: Yes, you are right. Thanks for the update.

File: Doc/library/zipfile.rst [hunk 5]
```diff
@@ -385,10 +385,10 @@ ZipFile Objects
 
 .. method:: ZipFile.writestr(zinfo_or_arcname, data[, compress_type])
 
-   Write the string *data* to the archive; *zinfo_or_arcname* is either the file
-   name it will be given in the archive, or a :class:`ZipInfo` instance.  If it's
-   an instance, at least the filename, date, and time must be given.  If it's a
-   name, the date and time is set to the current date and time.
+   `ZipFile.write()` writes a file into archive. `ZipFile.writestr()` writes a bytes data or a string.
```
Comment (serhiy-storchaka) [inline Doc/library/zipfile.rst]: Oh, I didn't mean that you should include my words verbatim. `ZipFile.write()` is documented in the description of `ZipFile.write()`.

The existing documentation is mostly good. The only problem is that the term "string" is used in wider meaning that includes byte strings, but this is not clear. Just document what actually does this method.

**Pull request closed without merging**